modules/drivers/druid/src/metabase/driver/druid.clj

```diff
@@ -29,11 +30,18 @@
   [_ query]
   (druid.qp/mbql->native query))
 
+(defn- add-timeout-to-query [query timeout]
+  (let [parsed (if (string? query)
+                 (json/parse-string query keyword)
+                 query)]
+    (assoc-in parsed [:context :timeout] timeout)))
+
 (defmethod driver/execute-reducible-query :druid
-  [_ query context respond]
+  [_ query {:keys [timeout] :as context} respond]
```

Comment:
It looks to me that the canonical way of getting timeout is `(qp.context/timeout context)`.

Reply:
Done